drgn/helpers/linux/fs.py

```diff
@@ -339,6 +339,8 @@ def for_each_file(task: Object) -> Iterator[Tuple[int, Object]]:
     :param task: ``struct task_struct *``
     :return: Iterator of (fd, ``struct file *``) tuples.
     """
+    if not task.files:
+        return
     fdt = task.files.fdt.read_()
```

Comment:
Good call, I've had to work around this a few times, too.

We can avoid an extra `/proc/kcore` read (and I guess be slightly less racy) if we read `task.files` into a value object and use that:

```suggestion
    files = task.files.read_()
    if not files:
        return
    fdt = files.fdt.read_()
```

Reply:
Yep, good point, it's updated now :)